sync/src/main/java/tech/pegasys/artemis/sync/AttestationManager.java

```diff
@@ -0,0 +1,145 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.sync;
+
+import static tech.pegasys.artemis.util.alogger.ALogger.STDOUT;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.artemis.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.artemis.datastructures.operations.AggregateAndProof;
+import tech.pegasys.artemis.datastructures.operations.Attestation;
+import tech.pegasys.artemis.service.serviceutils.Service;
+import tech.pegasys.artemis.statetransition.StateTransition;
+import tech.pegasys.artemis.statetransition.attestation.AttestationProcessingResult;
+import tech.pegasys.artemis.statetransition.attestation.ForkChoiceAttestationProcessor;
+import tech.pegasys.artemis.statetransition.events.BlockImportedEvent;
+import tech.pegasys.artemis.statetransition.events.ProcessedAggregateEvent;
+import tech.pegasys.artemis.storage.ChainStorageClient;
+import tech.pegasys.artemis.storage.events.SlotEvent;
+import tech.pegasys.artemis.util.async.SafeFuture;
+
+public class AttestationManager extends Service {
+  private static final Logger LOG = LogManager.getLogger();
+  private final EventBus eventBus;
+  private final ForkChoiceAttestationProcessor attestationProcessor;
+  private final PendingPool<Attestation> pendingAttestations;
+  private final FutureItems<Attestation> futureAttestations;
+
+  AttestationManager(
+      final EventBus eventBus,
+      final ForkChoiceAttestationProcessor attestationProcessor,
+      final PendingPool<Attestation> pendingAttestations,
+      final FutureItems<Attestation> futureAttestations) {
+    this.eventBus = eventBus;
+    this.attestationProcessor = attestationProcessor;
+    this.pendingAttestations = pendingAttestations;
+    this.futureAttestations = futureAttestations;
+  }
+
+  public static AttestationManager create(
+      final EventBus eventBus, final ChainStorageClient storageClient) {
+    final PendingPool<Attestation> pendingAttestations =
+        PendingPool.createForAttestations(eventBus);
+    final FutureItems<Attestation> futureAttestations =
+        new FutureItems<>(Attestation::getEarliestSlotForProcessing);
+    return new AttestationManager(
+        eventBus,
+        new ForkChoiceAttestationProcessor(storageClient, new StateTransition(false)),
+        pendingAttestations,
+        futureAttestations);
+  }
+
+  @Subscribe
+  @SuppressWarnings("unused")
+  private void onGossipedAttestation(final Attestation attestation) {
+    processAttestation(attestation);
+    // TODO: Should we post this if the attestation was invalid?
```

Comment:
I'd think we only want to post records if they're valid - did you mean to keep these TODO's in here?

Reply:
The behaviour hasn't changed from what it was previously, but I added the TODO's because it seemed extremely weird that we'd always consider it processed.  I lean towards only posting these events once we actually have processed the attestations which may be delayed.  Will follow up in another PR. https://pegasys1.atlassian.net/browse/BC-243 created for good measure.